gradoop-flink/src/main/java/org/gradoop/flink/model/impl/operators/tpgm/aggregation/AbstractTimeAggregateFunction.java

```diff
@@ -83,11 +98,45 @@ public PropertyValue getIncrement(TemporalElement element) {
     case TO:
       return PropertyValue.create(timeInterval.f1);
     default:
-      throw new IllegalArgumentException("Field " + field + " is not supported for time intervals" +
-        ".");
+      throw new IllegalArgumentException("Field " + field +
+        " is not supported for time intervals.");
     }
   }
 
+  /**
+   * Base aggregate function for min and max aggregations. Handles default behaviour and the
+   * logic of the comparison between the aggregate and increment value.
+   *
+   * @param aggregate The aggregate value.
+   * @param increment The increment value.
+   * @param comparison The function to apply the comparison between aggregate and increment value
+   * @return the aggregated value
+   */
+  PropertyValue compareAggregate(PropertyValue aggregate, PropertyValue increment,
+    BiFunction<PropertyValue, PropertyValue, PropertyValue> comparison) {
```

Comment:
I think that this name is not very fitting. This functions checks for default values and calls a function used to calculate the aggregate. You can either rename it (something like "callAggregateFunction" maybe?) or move this code to the `aggegate` function and let it call another function called `aggregateInternal` or something similar instead of using a `BiFunction`.

Reply:
Done - agree?